src/StatusBar.ts

```diff
@@ -41,32 +77,156 @@ export class StatusBar {
       },
     }
   }
+
+  onDidChangeActiveTextEditor(editor: vscode.TextEditor) {
+    if (editor && editor.document) {
+      const folder = vscode.workspace.getWorkspaceFolder(editor.document.uri)
+      if (folder && folder.name !== this.activeFolder) {
+        this._activeFolder = folder.name
+        this.updateActiveStatusItem()
+      }
+    }
+  }
+
   private enqueue(source: string, status: Status, details?: string) {
     this.queue = this.queue.filter(x => x.source !== source)
-    this.queue.unshift({
+    const item: QueueItem = {
       source,
       status,
       details,
-    })
-    this.updateStatus()
+    }
+    this.queue.unshift(item)
+    this.updateStatus(item)
   }
-  private updateStatus() {
-    for (const status of this.priorities) {
-      const queueItem = this.queue.find(item => item.status === status)
-      if (queueItem) {
-        this.render(queueItem)
-        break
+  private updateStatus(item: QueueItem) {
+    this.updateActiveStatusItem(item)
+    this.updateSummaryStatus()
+  }
+
+  private get activeFolder() {
+    if (!this._activeFolder && vscode.window.activeTextEditor) {
+      const folder = vscode.workspace.getWorkspaceFolder(vscode.window.activeTextEditor.document.uri)
+      if (folder) {
+        this._activeFolder = folder.name
+      }
+    }
+    return this._activeFolder
+  }
+
+  private updateActiveStatusItem(item?: QueueItem) {
+    if (item && this.activeFolder) {
+      if (item.source === this.activeFolder) {
+        this.render(item, StatusType.active)
+      }
+      return
+    }
+
+    // find the active item from the queue
+    const queueItem = this.activeFolder
+      ? this.queue.find(_item => _item.source === this.activeFolder)
+      : this.queue.length === 1
+      ? this.queue[0]
+      : undefined

```

Comment:
If this is not an explicit code style, i believe some indentation may be appropriate here :)

Reply:
even with indentation, I agree it could still be hard to read, sigh, breaking it up...